src/graphics/pyro/pyro.h

```diff
@@ -185,5 +161,185 @@ class CPyro
     float           m_resetAngle = 0.0f;
 };
 
+class CFlagCreatePyro : public CPyro
+{
+public:
+    CFlagCreatePyro(CObject *pObj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CFlagDeletePyro : public CPyro
+{
+public:
+    CFlagDeletePyro(CObject *pObj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CWaypointHitPyro : public CPyro
+{
+public:
+    CWaypointHitPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CFallPyro : public CPyro
+{
+public:
+    CFallPyro(CObject *obj);
+
+    //! Start of an object freight falling
+    void        AfterCreate() override;
+    //! Seeks an object to explode by the falling ball of bees
+    CObject*    FallSearchBeeExplo();
+    //! Fall of an object's freight
+    void        FallProgress(float rTime);
+    //! Indicates whether the fall is over
+    Error       IsEnded() override;
+    //! Makes a sound and unlocks the object after fall is done
+    void        AfterEnd() override;
+
+    bool EventProcess(const Event&) override;
+};
+
+class CResetPyro : public CPyro
+{
+public:
+    CResetPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CLostPyro : public CPyro
+{
+public:
+    CLostPyro(CObject *obj);
+    Error IsEnded() override;
+    void UpdateEffect() override;
+};
+
+class CWinPyro : public CPyro
+{
+public:
+    CWinPyro(CObject *obj);
+    Error IsEnded() override;
+    void UpdateEffect() override;
+};
+
+class CSpiderPyro : public CPyro
+{
+public:
+    CSpiderPyro(CObject *obj);
+    bool EventProcess(const Event& event) override;
+    void AfterCreate() override;
+};
+
+class CEggPyro : public CPyro
+{
+public:
+    CEggPyro(CObject *obj);
+    bool EventProcess(const Event& event) override;
+    void AfterCreate() override;
+};
+
+class CDeadGPyro : public CPyro
+{
+public:
+    CDeadGPyro(CObject *obj);
+    void AfterCreate() override;
+};
+
+class CDeadWPyro : public CPyro
+{
+public:
+    CDeadWPyro(CObject *obj);
+    void AfterCreate() override;
+};
+
+class CFindingPyro : public CPyro
+{
+public:
+    CFindingPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+};
+
+class CSquashPyro : public CPyro
+{
+public:
+    CSquashPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CFragVPyro : public CPyro
+{
+public:
+    CFragVPyro(CObject *obj);
+    bool EventProcess(const Event& event) override;
+    void AfterCreate() override;
+};
+
+class CBurnPyro : public CPyro
+{
+    bool m_organicBurn;
+public:
+    CBurnPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+
+    //! Starts a vehicle fire
+    void        BurnStart();
```

Comment:
BurnStart(), BurnAddPart(), BurnProgress() and BurnIsKeepPart() probably should not be public. Or are they used outside CBurnPyro?